### PR TITLE
Add Deployment Topologies docs

### PR DESCRIPTION
**Pull request by mheap**
### Description

Explain the different deployment topologies for KIC 3.0 + add Sidecar -> Gateway Discovery migration guide.

### Testing instructions

Preview links:

* https://deploy-preview-6389--kongdocs.netlify.app/kubernetes-ingress-controller/latest/production/deployment-topologies/
* https://deploy-preview-6389--kongdocs.netlify.app/kubernetes-ingress-controller/latest/production/deployment-topologies/gateway-discovery/
* https://deploy-preview-6389--kongdocs.netlify.app/kubernetes-ingress-controller/latest/production/deployment-topologies/db-backed/
* https://deploy-preview-6389--kongdocs.netlify.app/kubernetes-ingress-controller/latest/production/deployment-topologies/sidecar/

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/db-backed.md [hunk 1]
```diff
@@ -1,6 +1,29 @@
...
+title: Database Backed
 type: explanation
 purpose: |
   What is a DB-backed deployment and when should it be used?
 ---
+
+{:.important}
+> Database backed deployments should only be used in a small set of circumstances. We recommend using [Gateway Discovery](/kubernetes-ingress-controller/{{ page.release }}/production/deployment-topologies/gateway-discovery/) unless you've been otherwise advised by a member of the Kong team
+
+Database backed deployments are when {{ site.kic_product_name }} is used to update a Lua Control Plane that is attached to a Postgresql database. {{ site.base_gateway }} may be deployed in either [Traditional](/gateway/latest/production/deployment-topologies/traditional/) or [Hybrid](/gateway/latest/production/deployment-topologies/hybrid-mode/) mode.
```
[the remark this answers is not in the document]
Comment (programmer04) [reply app/_src/kubernetes-ingress-controller/production/deployment-topologies/db-backed.md:11]: PostgreSQL

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md [hunk 2]
```diff
@@ -4,3 +4,118 @@ type: explanation
...
+
+## Overview
+
+Sidecar deployment is the original method of deployment for {{ site.kic_product_name }}. Both the controller and {{ site.base_gateway }} are deployed in a single pod and each {{ site.base_gateway }} instance was managed by a different {{ site.kic_product_name }}.
+
+This is the simplest deployment method as everything is contained in a single deployment and the controller can communicate to {{ site.base_gateway }} on `localhost`.
+
+Sidecar deployments have been deprecated in favor of [Gateway Discovery](/kubernetes-ingress-controller/{{ page.release }}/production/deployment-topologies/gateway-discovery/) for multiple reasons:
+
+* Reduced resource usage as the number of {{ site.kic_product_name }} does not scale lineraly with {{ site.base_gateway }}
```
[the remark this answers is not in the document]
Comment (programmer04) [reply app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:19]: linearly

**Review by programmer04 — COMMENTED** [review]
lgtm

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/db-backed.md (hunk 1 above)
Comment (mheap) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/db-backed.md:11]:
```suggestion
Database backed deployments are when {{ site.kic_product_name }} is used to update a Lua Control Plane that is attached to a PostgreSQL database. {{ site.base_gateway }} may be deployed in either [Traditional](/gateway/latest/production/deployment-topologies/traditional/) or [Hybrid](/gateway/latest/production/deployment-topologies/hybrid-mode/) mode.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/db-backed.md [hunk 3]
```diff
@@ -1,6 +1,29 @@
...
+
+![Traditional Architecture Diagram](/assets/images/products/kubernetes-ingress-controller/topology/db-backed-traditional.png)
+
+{{ site.kic_product_name }} will send configuration to a random {{ site.base_gateway }} instance, which writes the configuration to the database. All other nodes read the configuration from the database.
+
+## Hybrid Mode
+
+Database backed Hybrid mode is similar to Traditional mode, but instead of every node reading from the database a single Control Plane is responsible for managing the configuration and distributing it to all active Data Planes.
+
+In Hybrid mode, {{ site.kic_product_name }} uses Gateway Discovery to find the Control Plane and send {{ site.base_gateway }} configuration to the Admin API. This configuration is persisted to the Postgresql database and transmitted to the Data Planes using the {{ site.base_gateway }} CP/DP protocol.
```
[the remark this answers is not in the document]
Comment (mheap) [reply app/_src/kubernetes-ingress-controller/production/deployment-topologies/db-backed.md:27]:
```suggestion
In Hybrid mode, {{ site.kic_product_name }} uses Gateway Discovery to find the Control Plane and send {{ site.base_gateway }} configuration to the Admin API. This configuration is persisted to the PostgreSQL database and transmitted to the Data Planes using the {{ site.base_gateway }} CP/DP protocol.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mheap) [reply app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:19]:
```suggestion
* Reduced resource usage as the number of {{ site.kic_product_name }} does not scale linearly with {{ site.base_gateway }}
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/index.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+---
+title: Overview
+type: explanation
+purpose: |
+  What deployment topologies are available and when should they be used?
+---
+
+{{ site.kic_product_name }} can be deployed in many different topologies depending on your needs. The primary ways to deploy {{ site.kic_product_name }} are:
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/index.md:8]:
```suggestion
{{ site.kic_product_name }} can be deployed in many different topologies depending on your needs. The primary methods to deploy {{ site.kic_product_name }} are:
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/index.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+1. [Gateway Discovery](/kubernetes-ingress-controller/{{ page.release }}/production/deployment-topologies/gateway-discovery/) - The recommended way to deploy {{ site.kic_product_name }}.
+1. [Database Backed with Gateway Discovery](/kubernetes-ingress-controller/{{ page.release }}/production/deployment-topologies/db-backed/) - Gateway Discovery with an additional Lua Control Plane and Postgres database.
+1. [Traditional (Sidecar)](/kubernetes-ingress-controller/{{ page.release }}/production/deployment-topologies/sidecar/) - Run {{ site.kic_product_name }} as a sidecar container in the same pod as {{ site.base_gateway }} (legacy deployment).
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/index.md:12]:
```suggestion
1. [Traditional (Sidecar)](/kubernetes-ingress-controller/{{ page.release }}/production/deployment-topologies/sidecar/) - Run {{ site.kic_product_name }} as a sidecar container in the same Pod as {{ site.base_gateway }} (legacy deployment).
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/gateway-discovery.md [hunk 6]
```diff
@@ -4,3 +4,18 @@ type: explanation
...
+
+It allows you to manage many {{ site.base_gateway }} instances with one {{ site.kic_product_name }}, providing lower resource usage compared to [sidecar](/kubernetes-ingress-controller/{{ page.release }}/production/deployment-topologies/sidecar/) deployments.
+
+## How It Works
+
+{:.note}
+> You don't need to configure this manually due to the default value of `gatewayDiscovery.generateAdminApiService=true` in the Helm chart. The following information is for educational purposes only.
+
+When {{ site.kic_product_name }} starts running it looks for services that match the name in the `--kong-admin-svc` flag. This value is controlled by the `gatewayDiscovery.adminApiService.namespace` and `gatewayDiscovery.adminApiService.name` values in the Helm chart. One the service is found, {{ site.kic_product_name }} fetches a list of [EndpointSlices](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/) containing all {{ site.base_gateway }} instances.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/gateway-discovery.md:17]:
```suggestion
When {{ site.kic_product_name }} starts running it looks for services that match the name in the `--kong-admin-svc` flag. This value is controlled by the `gatewayDiscovery.adminApiService.namespace` and `gatewayDiscovery.adminApiService.name` values in the Helm chart. Once the service is found, {{ site.kic_product_name }} fetches a list of [EndpointSlices](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/) containing all {{ site.base_gateway }} instances.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/db-backed.md [hunk 7]
```diff
@@ -1,6 +1,29 @@
...
+
+Both [Gateway Discovery](/kubernetes-ingress-controller/{{ page.release }}/production/deployment-topologies/gateway-discovery/) and [Sidecar](/kubernetes-ingress-controller/{{ page.release }}/production/deployment-topologies/sidecar/) deployments can be used alongside a database. If you're starting a greenfield project today we recommend using Gateway Discovery.
+
+## Traditional Mode
+
+Traditional mode is when every {{ site.base_gateway }} instance acts as both a Control Plane and a Data Plane. All nodes connect to the database and load the latest configuration at a regular interval.
+
+![Traditional Architecture Diagram](/assets/images/products/kubernetes-ingress-controller/topology/db-backed-traditional.png)
+
+{{ site.kic_product_name }} will send configuration to a random {{ site.base_gateway }} instance, which writes the configuration to the database. All other nodes read the configuration from the database.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/db-backed.md:21]:
```suggestion
{{ site.kic_product_name }} sends configuration to a random {{ site.base_gateway }} instance, which writes the configuration to the database. All other nodes read the configuration from the database.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md [hunk 8]
```diff
@@ -4,3 +4,118 @@ type: explanation
 purpose: |
   What is Sidecar Mode? Why is it deprecated?
 ---
+
+{:.important}
+> Sidecar deployments are officially supported, but discouraged. We recommend using [Gateway Discovery](/kubernetes-ingress-controller/{{ page.release }}/production/deployment-topologies/gateway-discovery/) going forwards.
+
+## Overview
+
+Sidecar deployment is the original method of deployment for {{ site.kic_product_name }}. Both the controller and {{ site.base_gateway }} are deployed in a single pod and each {{ site.base_gateway }} instance was managed by a different {{ site.kic_product_name }}.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:13]:
```suggestion
Sidecar deployment is the original method of deployment for {{ site.kic_product_name }}. Both the controller and {{ site.base_gateway }} are deployed in a single Pod and each {{ site.base_gateway }} instance was managed by a different {{ site.kic_product_name }}.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md [hunk 9]
```diff
@@ -4,3 +4,118 @@ type: explanation
...
+
+This is the simplest deployment method as everything is contained in a single deployment and the controller can communicate to {{ site.base_gateway }} on `localhost`.
+
+Sidecar deployments have been deprecated in favor of [Gateway Discovery](/kubernetes-ingress-controller/{{ page.release }}/production/deployment-topologies/gateway-discovery/) for multiple reasons:
+
+* Reduced resource usage as the number of {{ site.kic_product_name }} instances does not scale linearly with {{ site.base_gateway }}
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:19]:
```suggestion
* Reduce resource usage as the number of {{ site.kic_product_name }} instances does not scale linearly with {{ site.base_gateway }}.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md [hunk 10]
```diff
@@ -4,3 +4,118 @@ type: explanation
...
+* Reduced load on the Kubernetes API server. There are fewer clients, and no thrashing behaviour as multiple controllers argue of the `programmed` state of a resource
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:20]:
```suggestion
* Reduce load on the Kubernetes API server. There are fewer clients, and no thrashing behaviour as multiple controllers argue of the `programmed` state of a resource.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md [hunk 11]
```diff
@@ -4,3 +4,118 @@ type: explanation
...
+* The ability to scale {{ site.kic_product_name }} and {{ site.base_gateway }} independently as needed.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:21]:
```suggestion
* Scale {{ site.kic_product_name }} and {{ site.base_gateway }} independently as needed.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md [hunk 12]
```diff
@@ -4,3 +4,118 @@ type: explanation
...
+
+![Sidecar Architecture Diagram](/assets/images/products/kubernetes-ingress-controller/topology/sidecar.png)
+
+## Migrating to Gateway Discovery
+
+If you see two containers running in the same pod, it's likely that you're running {{ site.kic_product_name }} in Sidecar mode.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:27]:
```suggestion
If you see two containers running in the same Pod, it's likely that you're running {{ site.kic_product_name }} in Sidecar mode.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md [hunk 13]
```diff
@@ -4,3 +4,118 @@ type: explanation
...
+ingressController:
+  enabled: false
+
+admin:
+  enabled: true
+  type: ClusterIP
+  clusterIP: None
+```
+
+The new Proxy pod won't come online as there is no available configuration:
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:63]:
```suggestion
The new Proxy Pod won't come online as there is no available configuration.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md [hunk 14]
```diff
@@ -4,3 +4,118 @@ type: explanation
...
+
+deployment:
+  kong:
+    enabled: false
+```
+
+{:.note}
+> `kong-kong-admin` is the default name if your release is called `kong`. Run `kubectl get services -n kong` to find the correct name
+
+Create a new `controller` deployment using Helm:
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:87]:
```suggestion
Create a new `controller` deployment using Helm.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md [hunk 15]
```diff
@@ -4,3 +4,118 @@ type: explanation
...
+
+```bash
+helm install controller kong/kong --values ./values-controller.yaml -n kong
+```
+
+The new pods will not come online as the controller can't access the Admin API for the original Proxy pod. Delete the old Proxy pod to allow Gateway Discovery to work.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:93]:
```suggestion
The new Pods do not come online as the controller can't access the Admin API for the original Proxy Pod. Delete the old Proxy Pod to allow Gateway Discovery to work.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md [hunk 16]
```diff
@@ -4,3 +4,118 @@ type: explanation
...
+
+{:.important}
+> There may be a small amount of downtime of up to three seconds between the pod being deleted and new proxy pods receiving a configuration
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:96]:
```suggestion
> There may be a small amount of downtime of up to three seconds between the Pod being deleted and new proxy pods receiving a configuration.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md [hunk 17]
```diff
@@ -4,3 +4,118 @@ type: explanation
...
+
+```bash
+kubectl get pods -n kong
+```
+
+Find the pod with two containers. This is the old Sidecar topology that we need to delete.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:102]:
```suggestion
Find the Pod with two containers. This is the old Sidecar topology that needs to be deleted.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md [hunk 18]
```diff
@@ -4,3 +4,118 @@ type: explanation
...
+
+```text
+NAME                               READY   STATUS    RESTARTS   AGE
+kong-kong-7f5bddf88c-6cnlq         2/2     Running   0          2m
+```
+
+Delete the pod:
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:109]:
```suggestion
Delete the Pod.
```

File: app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md [hunk 19]
```diff
@@ -4,3 +4,118 @@ type: explanation
...
+
+```bash
+kubectl delete pod -n kong kong-kong-7f5bddf88c-6cnlq
+```
+
+At this point you will have two pods running. `controller-kong` contains {{ site.kic_product_name }} and `kong-kong` contains {{ site.base_gateway }}
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/production/deployment-topologies/sidecar.md:115]:
```suggestion
At this point there are two Pods running. `controller-kong` contains {{ site.kic_product_name }} and `kong-kong` contains {{ site.base_gateway }}
```

**Review by Rajakavitha1 — APPROVED** [review]
LGTM!

**Review by programmer04 — APPROVED** [review]

**Pull request merged**